flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/sessionjob/FlinkSessionJobReconciler.java

```diff
@@ -119,38 +116,39 @@ public void reconcile(FlinkSessionJob flinkSessionJob, Context context) throws E
                 if (desiredJobState == JobState.RUNNING) {
                     LOG.info("Upgrading/Restarting running job, suspending first...");
                 }
-                stateAfterReconcile = suspendJob(flinkSessionJob, upgradeMode, effectiveConfig);
+                stateAfterReconcile = suspendJob(flinkSessionJob, upgradeMode, deployedConfig);
             }
             if (currentJobState == JobState.SUSPENDED && desiredJobState == JobState.RUNNING) {
                 if (upgradeMode == UpgradeMode.STATELESS) {
-                    submitAndInitStatus(flinkSessionJob, effectiveConfig, null);
+                    submitAndInitStatus(flinkSessionJob, deployedConfig, null);
```

Comment:
I think here we should use `getDeployConfig` to submit a new job instead of using `getObserveConfig`?

Reply:
I think this is correct, we need to get the config that is running on the cluster at the moment. Which is the observeConfig